### ISSUE TITLE
Not working on Android 14

### ISSUE THREAD

**Issue by fredx100** [anevero/sms_my_gps#4]
As far as I can tell, this isn't working on android 14 (on a pixel 7a, at least). It could be that I've set it up incorrectly. However, I setup certain numbers with their prefixes. Then, receiving messages from those numbers (with the appropriate prefixes) doesn't appear to trigger any responses. The service appears to be running (it's present on my notification bar).

Is this expected?

**Comment by anevero** [comment]
Hi @fredx100, I have verified the app is working on a Pixel 7 Pro with Android 14 (although some improvements for recent Android versions are possible).

Could you please confirm if you have one or two SIM cards in your phone? If you have two SIM cards, is any of them selected as default for SMS? SMS sending code in the app [relies on the fact](https://github.com/anevero/sms_my_gps/blob/bf1877aadb7bb8be4343567ac813da44c4048a70/app/src/main/java/com/github/anevero/sms_my_gps/events/SMSSender.java#L27) there is a default SIM for SMS. If the default SIM is not set, the app will silently (without any user-visible errors) abort sending messages.

**Comment by fredx100** [comment]
Hi @anevero, that's great news that it's possible. I only have one sim card. Could the issue be that the received messages are sent by RCS rather than SMS? Are there any prohibited characters for the message prefix?

**Comment by anevero** [comment]
@fredx100 yes, the app doesn't support RCS messages. Android handles these messages differently (and the way to handle them depends on the messaging app and phone manufacturer). Does it work if you send an SMS, not an RCS message?

**Comment by fredx100** [comment]
I've tested with SMS* and still no success. Could it be that I'm just not getting a GPS or network lock in time? I don't use google location services. Is there a way to see, in the gui or a log, whether smsgps has detected that it's received a request or would I need to debug to see that?

\* I deactivated RCS on both the other phone, which was sending the request, and my own phone, which has sms_my_gps on it, just for good measure!

**Comment by fredx100** [comment]
I looked a bit further into this this afternoon. I tested it out when my phone already had a GPS lock in another app but still without success. As the phone already had a GPS location I imagine the issue can't be that it's timing out while trying to get one.

FYI I've also set the required accuracy to be the loosest (200m) for testing purposes. I also checked the permissions for the app: it has permission to access the location "when the app is running", and the app is visible in the notification bar.

Let me know if there's anything else I can try which might be useful.

**Comment by anevero** [comment]
Hi @fredx100 , sorry for the late response. Unfortunately, the application doesn't save the history of received events, so it's impossible to see them in the UI.

I believe you will need to build and deploy the application via Android Studio to see any logs. If I remember correctly, it has some debug logging about SMS messages it received, and location information it fetched.

**Comment by fredx100** [comment]
Hi @anevero, sorry for my even-later response! I finally found time to look into this and the explanation is quite simple. I'd manually entered the ListItem's telephone number with a leading 0 rather than the country code, and the SmsListener then attempts to match this against the number bundled with the sms data and doesn't complete the match.

I've sent a somewhat trivial pull-request which fixes this for me by trimming any leading zero from a ListItem's number prior to matching it against the end of the sms' originating number. I will take no offence if you think there's a better implementation or approach!
